platform/wayland/cog-view-wl.c

```diff
@@ -278,7 +274,7 @@ static void
 cog_wl_view_request_frame(CogWlView *view)
 {
     CogWlPlatform *platform = (CogWlPlatform *) cog_platform_get();
-    CogWlViewport *viewport = COG_WL_VIEWPORT(platform->viewport);
+    CogWlViewport *viewport = COG_WL_VIEWPORT(cog_view_get_viewport((CogView *) view));
```

Comment:
Same here: use `g_autoptr` for the `viewport` pointer, and it needs to be null-checked.